third_party/forked/reflect/deep_equal.go

```diff
@@ -156,7 +156,10 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 		}
 		return true
 	case reflect.Slice:
-		if (v1.IsNil() || v1.Len() == 0) != (v2.IsNil() || v2.Len() == 0) {
+		// go1.5: simplified if to work around https://github.com/golang/go/issues/12266
```

Comment:
You can argue that they should be removed once the current version of Go is 1.5.1. But for as long as the stable is 1.5, not including this patch will mean that you can't use Kubernetes with the current stable version, not even for testing/development purposes.